packages/smart-contracts/src/contracts/EthConversionProxy.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ChainlinkConversionPath.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title EthConversionProxy
+ * @notice This contract convert from chainlink then swaps ETH
+ *         before paying a request thanks to a conversion payment proxy
+ *         the dependance with ReentrancyGuard is required to perform
+ *         "transferExactEthWithReferenceAndFee" of the eth-fee-proxy contract
+  */
+contract EthConversionProxy is ReentrancyGuard {
+  address public paymentProxy;
+  ChainlinkConversionPath public chainlinkConversionPath;
+  address public nativeTokenHash;
+
+  constructor(address _paymentProxyAddress, address _chainlinkConversionPathAddress, address _nativeTokenHash) {
+    paymentProxy = _paymentProxyAddress;
+    chainlinkConversionPath = ChainlinkConversionPath(_chainlinkConversionPathAddress);
+    nativeTokenHash = _nativeTokenHash;
+  }
+
+  // Event to declare a conversion with a reference
+  event TransferWithConversionAndReference(
+    uint256 amount,
+    address currency,
+    bytes indexed paymentReference,
+    uint256 feeAmount,
+    uint256 maxRateTimespan
+  );
+  
+  // Event to declare a transfer with a reference
+  event TransferWithReferenceAndFee(
+    address to,
+    uint256 amount,
+    bytes indexed paymentReference,
+    uint256 feeAmount,
+    address feeAddress
+  );
+
+  /**
+   * @notice Performs an ETH transfer with a reference computing the payment amount based on the request amount
+   * @param _to Transfer recipient of the payement
+   * @param _requestAmount Request amount
+   * @param _path Conversion path
+   * @param _paymentReference Reference of the payment related
+   * @param _feeAmount The amount of the payment fee
+   * @param _feeAddress The fee recipient
+   * @param _maxRateTimespan Max time span with the oldestrate, ignored if zero
+   */
+  function transferWithReferenceAndFee(
+    address _to,
+    uint256 _requestAmount,
+    address[] calldata _path,
+    bytes calldata _paymentReference,
+    uint256 _feeAmount,
+    address _feeAddress,
+    uint256 _maxRateTimespan
+  )
+    external
+    payable
+  {
+    require(
+      _path[_path.length - 1] == nativeTokenHash,
```

Comment:
what happens if _path is empty?

Reply:
`_path.length - 1` will throw (underflow)

Reply:
which is OK I guess... thanks!